functions/src/importMentors/index.js

```diff
@@ -0,0 +1,84 @@
+const parse = require('csv-parse');
+const { XMLHttpRequest } = require('xmlhttprequest');
+const { db, functions, admin } = require('../utils/firestore');
+
+const fieldNamesOrdered = [
+  'timestamp',
+  'email',
+  'firstName',
+  'lastName',
+  'mentorOrExpo',
+  'yearsOfExperience',
+  'areaOfMentorship',
+  'isStayingOvernight',
+  'tShirtSize',
+  'diet',
+  'source',
+  'womenInTech',
+];
+
+// cloud function for submitting hacker application
+exports.importMentors = functions.https.onRequest((request, response) => {
+  // set origin and headers to allow CORS for local testing
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+  response.set('Access-Control-Allow-Origin', '*');
+  response.set('Access-Control-Allow-Headers', 'Content-Type, crossDomain');
+  response.set('Access-Control-Allow-Credentials', 'true');
+  response.set('Access-Control-Allow-Methods', 'POST');
+
+  if (request.method === 'OPTIONS') {
+    // Send response to OPTIONS requests
+    response.set('Access-Control-Allow-Headers', 'Authorization');
+    response.set('Access-Control-Max-Age', '3600');
+    return response.status(204).send('');
+  }
+
+  if (request.method !== 'POST') {
+    return response.status(400).send('Unrecognized method'); // database error
+  }
+
+  // ensure request.body is an object, not a string
+  const data = typeof request.body !== 'object' ? JSON.parse(request.body) : request.body;
+
+  const clientToken = request.header('Authorization');
+
+  return admin.auth().verifyIdToken(clientToken).then((decodedToken) => {
+    const { uid } = decodedToken;
+    return db.collection('admins').doc(uid).get();
+  }).then((adminDoc) => {
+    if (!adminDoc) {
+      throw new Error('Not an admin');
+    }
+
+    const { csvUrl } = data;
+
+    const xhr = new XMLHttpRequest();
+    xhr.responseType = 'text';
+    xhr.onload = function (event) {
+      const { responseText } = xhr;
+      parse(responseText, (err, output) => {
+        if (err) {
+          return response.status(500).send(err);
+        }
+
+        const batch = db.batch();
+
+        output.forEach((row) => {
```

Comment:
Instead of using the `timestamp` from the CSV, can we use `const timestamp = Date.now()` for every iteration of this loop (for consistency, and to avoid updating existing code which assumes timestamp is in Unix time)? The CSV is ordered by timestamp it seems, so the relative order of the mentors should stay the same.